src/averageAges.js

```diff
@@ -53,9 +70,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @param {boolean} onlyWithSon - optional
  *
  * @return {number}
+ *
+ *
+ *
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childrens = people.filter(person =>
+    onlyWithSon
+      ? people.some(item => item.name === person.mother && person.sex === 'm')
+      : people.some(item => item.name === person.mother));
+
+  const diff = childrens.map(child => {
```

Comment:
Consider using more descriptive variable names, like `ageDiff` instead of `diff`. 🤔